torch/csrc/lazy/backend/backend_device.cpp

```diff
@@ -56,7 +56,9 @@ c10::Device backendDeviceToAtenDevice(const BackendDevice& device) {
 
 c10::optional<BackendDevice> GetBackendDevice(const at::Tensor& tensor) {
   if (auto lt = TryGetLtcTensor(tensor)) {
-    return lt.GetDevice();
+    if (*lt) {
```

Comment:
this looks a little questionable!

Reply:
hah yea, you caught me. it is legit,  i think, but indicates a necessary refactor/cleanup.

old world:
LazyTensor::operator bool() -- returns false if LazyTensor.data is null

new world:
LazyTensorPtr::operator bool() -- returns false if LazyTensorPtr is uninitialized
LazyTensor::operator bool() -- returns false if LazyTensor.data is null

we should delete the 'Data' container class inside LazyTensor and rely entirely on unitialized LazyTensorPtr as the only way to represent a 'null' LazyTensor.

But to decouple that big change, I wanted to just assume Data() is never allowed to be null anymore but otherwise leave the API as-is.  I think I can make this change now (assert LazyTensor.data() is never null) and then I could remove this check.